[lldb/Interpreter] Implement ScriptedFrameProvider{,Python}Interface

medismailben opened the pull request:
This patch implements the base and python interface for the
ScriptedFrameProvider class.

This is necessary to call python APIs from the ScriptedFrameProvider
that will come in a follow-up.

Signed-off-by: Med Ismail Bennani <ismail@bennani.ma>

bulbazord reviewed (commented):
I don't see anything egregious. Deferring to Jonas's judgement.

JDevlieghere reviewed (approved):
LGTM modulo nits.

medismailben committed:
[lldb/Interpreter] Implement ScriptedFrameProvider{,Python}Interface

This patch implements the base and python interface for the
ScriptedFrameProvider class.

This is necessary to call python APIs from the ScriptedFrameProvider
that will come in a follow-up.

Signed-off-by: Med Ismail Bennani <ismail@bennani.ma>
